GameOfRunes/GameOfRunes/GameEngine/GameEngine+SystemDelegate.swift

```diff
@@ -8,9 +8,8 @@
 
 import GameplayKit
 
-class SystemDelegate {
-    private weak var gameEngine: GameEngine?
-    private var systems = [ComponentType: System]()
+/** Handles all system related logic */
+extension GameEngine {
```

Comment:
if you wanna do extension then we should do a protocol - extension like what i did, then hold a reference to it as well

Reply:
A weak ref?

Reply:
yeah, if not theres a memory leak since its holding onto `self`

Reply:
Hmm i feel a delegate is not required. Since this would mean we need boiler plate code in the main game engine.
E.g. func dosth() {
    systemDelegate?.dosth()
}

Reply:
cuz rn its just not a delegate, it's just an extension to perform system logic but thats ok I guess